core/src/main/java/org/elasticsearch/common/network/NetworkModule.java

```diff
@@ -381,6 +395,37 @@ public void registerTaskStatus(String name, Writeable.Reader<? extends Task.Stat
         namedWriteableRegistry.register(Task.Status.class, name, reader);
     }
 
+    /**
+     * Register an allocation command.
+     * <p>
+     * This lives here instead of the more aptly named ClusterModule because the Transport client needs these to be registered.
+     * </p>
+     * @param reader the reader to read it from a stream
+     * @param parser the parser to read it from XContent
+     * @param commandName the names under which the command should be parsed. The {@link ParseField#getPreferredName()} is special because
+     *        it is the name under which the command's reader is registered.
+     */
+    public <T extends AllocationCommand> void registerAllocationCommand(Writeable.Reader<T> reader, AllocationCommand.Parser<T> parser,
+            ParseField commandName) {
+        // TODO Refactor when #17653 is merged
+        Tuple<ParseField, AllocationCommand.Parser<?>> parseFieldParserTuple = new Tuple<>(commandName, parser);
+        for (String name: commandName.getAllNamesIncludedDeprecated()) {
+            Tuple<ParseField, AllocationCommand.Parser<?>> previousValue = allocationCommands.putIfAbsent(name, parseFieldParserTuple);
+            if (previousValue != null) {
+                throw new IllegalArgumentException("Allocation command parser [" + previousValue.v2() + "] already registered for name [" +
+                        name + "] while trying to register [" + parser + "]");
+            }
+        }
+        namedWriteableRegistry.register(AllocationCommand.class, commandName.getPreferredName(), reader);
+    }
+
+    /**
+     * The registry of allocation command parsers.
+     */
+    public AllocationCommandRegistry getAllocationCommandRegistry() {
```

Comment:
I was not talking about moving the registration of the defaults but the `registerAllocationCommand` method.